### PR TITLE
[HOLD] review dao.c

### PR DESCRIPTION
**Pull request by doitian**
This PR is used to show the review comments inline.

### CODE REVIEW

File: c/dao.c [hunk 1]
```diff
@@ -295,6 +295,7 @@ static int calculate_dao_input_capacity(size_t input_index,
  * be met:
  *
  * * withdrawing cell uses the same lock script as deposited cell
+ * Is it an anti-pattern to let user reuse the key?
```
Comment (xxuejie) [inline c/dao.c]: Here I'd say it's more about changing the cell data to change the cell's status, all the other cell properties are inherited. This is not really about reusing the key, I'd see it's more about the unchanged part in a cell.

File: c/dao.c [hunk 2]
```diff
@@ -396,6 +397,7 @@ int main() {
   }
   script_seg.ptr = (uint8_t *)script;
   script_seg.size = len;
+  // Check len <= SCRIPT_SIZE
```
Comment (xxuejie) [inline c/dao.c]: Already done.

File: c/dao.c [hunk 3]
```diff
@@ -434,6 +436,7 @@ int main() {
           (memcmp(script_hash, current_script_hash, HASH_SIZE) == 0)) {
         dao_input = 1;
       }
+    // Capacity must exist for all inputs?
```
Comment (xxuejie) [inline c/dao.c]: I think this is something left from previous implementations. We can remove it.

File: c/dao.c (hunk 1 above)
Comment (doitian) [reply c/dao.c]: The lock has to be the same. Which means the out cell uses a lock which has already been unlocked once.

File: c/dao.c (hunk 3 above)
Comment (xxuejie) [reply c/dao.c]: Done in #65

File: c/dao.c (hunk 1 above)
Comment (xxuejie) [reply c/dao.c]: Done in #65

**Pull request closed without merging**